src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -727,21 +701,17 @@ void c_typecheck_baset::typecheck_expr_operands(exprt &expr)
      expr.get(ID_statement)==ID_function_call)
   {
     // don't do function operand
-    assert(expr.operands().size()==2);
-
-    typecheck_expr(expr.op1()); // arguments
+    typecheck_expr(to_binary_expr(expr).op1()); // arguments
```

Comment:
`to_side_effect_expr_function_call`

Reply:
Again -- it's not necessarily the case that at this point, all class invariants of this class have been established!